pkg/sql/colexec/external/external.go

```diff
@@ -667,22 +677,31 @@ func ScanCsvFile(param *ExternalParam, proc *process.Process) (*batch.Batch, err
 }
 
 func getFileService(param *ExternalParam, dir string) (fileservice.FileService, error) {
-	c := fileservice.Config{
-		Name:    defines.LocalFileServiceName,
-		Backend: "DISK",
-		DataDir: dir,
+	service := param.extern.FileService
+	path, err := fileservice.ParsePathAtService(param.extern.Filepath, "")
+	if err != nil {
+		return nil, err
 	}
-	if param.extern.ScanType == tree.S3 {
-		c.Name = defines.S3FileServiceName
-		c.Backend = "S3"
-	} else if strings.HasPrefix(param.extern.Filepath, "etl:") {
-		c.Name = defines.ETLFileServiceName
-		c.Backend = "DISK-ETL"
+	if path.Service == "" {
+		path.Service = service.Name()
+	}
+	switch path.Service {
+	case "LOCAL":
```

Comment:
The LOCAL file service is global and its root path is configured in the config file. You should never change it.
Just use GetForETL. If GetForETL does not return what you want, fix it.